sstables/compaction.cc

```diff
@@ -414,8 +416,10 @@ class compaction {
     }
 
     uint64_t partitions_per_sstable() const {
+        auto& cs = _cf.get_compaction_strategy();
+        auto adjusted_estimated_partitions = cs.adjust_partition_estimate(_ms_metadata, _estimated_partitions);
         uint64_t estimated_sstables = std::max(1UL, uint64_t(ceil(double(_info->start_size) / _max_sstable_size)));
-        return ceil(double(_estimated_partitions) / estimated_sstables);
+        return ceil(double(adjusted_estimated_partitions) / estimated_sstables);
```

Comment:
there are two adjustments going on here, one possibly by time window if cs == TWCS, and another by data size (total input size / max sstable size). That seems incorrect to me. estimated_sstables means maximum amount of segregation by data size, now it means max # of estimated sstables. cs.adjust_partition_estimate() has its own notion of estimated sstables, and its return value is the result of estimated partitions divided by that notion of estimated sstables. I think partitions_per_sstable() should return the lowest value of adjusted_estimated_partitions and (_estimated_partitions / estimated_sstables), the latter is like the maximum estimated partitions.

Reply:
BTW, I think we later should move data segregation by data size into a consumer of its own, which would be used only by strategies that limit sstable sizes like LCS. So partitions_per_sstable() implementation will only be a call to cs.adjust_partition_estimate().

Reply:
Good point I will change.